src/batchproof_container.cpp

```diff
@@ -171,75 +195,93 @@ void BatchProofContainer::batch_lelantus() {
+                            itr->first.first.first,
+                            itr->first.first.second,
+                            coins);
+                    anonymity_set.reserve(coins.size());
+                    for (auto& coin : coins)
+                        anonymity_set.emplace_back(coin.getValue());
+                } else {
+                    int coinGroupId = itr->first.first.first % (CENT / 1000);
+                    int64_t intDenom = (itr->first.first.first - coinGroupId);
+                    intDenom *= 1000;
+                    sigma::CoinDenomination denomination;
+                    sigma::IntegerToDenomination(intDenom, denomination);
+
+                    std::vector<GroupElement> coins;
+                    sigma::CSigmaState* sigmaState = sigma::CSigmaState::GetState();
+                    sigmaState->GetAnonymitySet(
+                            denomination,
+                            coinGroupId,
+                            true,
+                            coins);
+
+                    anonymity_set.reserve(coins.size());
+                    for (auto& coin : coins)
+                        anonymity_set.emplace_back(coin + params->get_h1() * intDenom);
+                }
 
-        size_t m = itr.second.size();
-        std::vector<Scalar> serials;
-        serials.reserve(m);
-        std::vector<size_t> setSizes;
-        setSizes.reserve(m);
-        std::vector<lelantus::SigmaExtendedProof> proofs;
-        proofs.reserve(m);
-        std::vector<Scalar> challenges;
-        challenges.reserve(m);
-
-        for (auto& proofData : itr.second) {
-            serials.emplace_back(proofData.serialNumber);
-            setSizes.emplace_back(proofData.anonymitySetSize);
-            proofs.emplace_back(proofData.lelantusSigmaProof);
-            challenges.emplace_back(proofData.challenge);
-        }
+                size_t m = itr->second.size();
+                std::vector<Scalar> serials;
+                serials.reserve(m);
+                std::vector<size_t> setSizes;
+                setSizes.reserve(m);
+                std::vector<lelantus::SigmaExtendedProof> proofs;
+                proofs.reserve(m);
+                std::vector<Scalar> challenges;
+                challenges.reserve(m);
+
+                for (auto& proofData : itr->second) {
+                    serials.emplace_back(proofData.serialNumber);
+                    setSizes.emplace_back(proofData.anonymitySetSize);
+                    proofs.emplace_back(proofData.lelantusSigmaProof);
+                    challenges.emplace_back(proofData.challenge);
+                }
 
-        lelantus::SigmaExtendedVerifier sigmaVerifier(params->get_g(), params->get_sigma_h(), params->get_sigma_n(),
-                                            params->get_sigma_m());
+                lelantus::SigmaExtendedVerifier sigmaVerifier(params->get_g(), params->get_sigma_h(), params->get_sigma_n(),
```

Comment:
I think you can move this outside of loop

Reply:
done